File: abc138/abc138_c.cs
// https://atcoder.jp/contests/abc138/submissions/7037895

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = int.Parse(Console.ReadLine());
        var v = Console.ReadLine().Split(' ').Select(x => double.Parse(x)).ToArray();
        Array.Sort(v);
        Array.Reverse(v);

        var len = v.Length;
        while (len > 1)
        {
            v[len - 2] = (v[len - 2] + v[len - 1]) / 2.0;
            len--;
        }
        Console.WriteLine($"{v[0]}");
    }
}

File: abc/abc138/abc138_c.fs
// https://atcoder.jp/contests/abc138/submissions/11992529

let N = stdin.ReadLine() |> int

let v =
    stdin.ReadLine().Split()
    |> Array.map float
    |> Array.sortDescending

let rec solver l =
    if l <= 1 then ()
    else
        v.[l - 2] <- (v.[l - 2] + v.[l - 1]) / 2.0
        solver (l - 1)

solver v.Length
v.[0] |> stdout.WriteLine
